package-lock.json

```diff
@@ -17300,6 +17704,135 @@
+          "requires": {
+            "domelementtype": "^2.3.0",
+            "domhandler": "^5.0.2",
+            "entities": "^4.2.0"
+          }
+        },
+        "domhandler": {
+          "version": "5.0.3",
+          "resolved": "https://registry.npmjs.org/domhandler/-/domhandler-5.0.3.tgz",
+          "integrity": "sha512-cgwlv/1iFQiFnU96XXgROh8xTeetsnJiDsTc7TYCLFd9+/WNkIqPTxiM/8pSd8VIrhXGTf1Ny1q1hquVqDJB5w==",
+          "requires": {
+            "domelementtype": "^2.3.0"
+          }
+        },
+        "domutils": {
+          "version": "3.1.0",
+          "resolved": "https://registry.npmjs.org/domutils/-/domutils-3.1.0.tgz",
+          "integrity": "sha512-H78uMmQtI2AhgDJjWeQmHwJJ2bLPD3GMmO7Zja/ZZh84wkm+4ut+IUnUdRa8uCGX88DiVx1j6FRe1XfxEgjEZA==",
+          "requires": {
+            "dom-serializer": "^2.0.0",
+            "domelementtype": "^2.3.0",
+            "domhandler": "^5.0.3"
+          }
+        },
+        "fs-extra": {
+          "version": "10.1.0",
+          "resolved": "https://registry.npmjs.org/fs-extra/-/fs-extra-10.1.0.tgz",
+          "integrity": "sha512-oRXApq54ETRj4eMiFzGnHWGy+zo5raudjuxN0b8H7s/RU2oW0Wvsx9O0ACRN/kRq9E8Vu/ReskGB5o3ji+FzHQ==",
+          "requires": {
+            "graceful-fs": "^4.2.0",
+            "jsonfile": "^6.0.1",
+            "universalify": "^2.0.0"
+          }
+        },
+        "htmlparser2": {
+          "version": "9.1.0",
+          "resolved": "https://registry.npmjs.org/htmlparser2/-/htmlparser2-9.1.0.tgz",
+          "integrity": "sha512-5zfg6mHUoaer/97TxnGpxmbR7zJtPwIYFMZ/H5ucTlPZhKvtum05yiPK3Mgai3a0DyVxv7qYqoweaEd2nrYQzQ==",
+          "requires": {
+            "domelementtype": "^2.3.0",
+            "domhandler": "^5.0.3",
+            "domutils": "^3.1.0",
+            "entities": "^4.5.0"
+          }
+        }
+      }
+    },
+    "@emnapi/core": {
+      "version": "1.3.1",
+      "resolved": "https://registry.npmjs.org/@emnapi/core/-/core-1.3.1.tgz",
+      "integrity": "sha512-pVGjBIt1Y6gg3EJN8jTcfpP/+uuRksIo055oE/OBkDNcjZqVbfkWCksG1Jp4yZnj3iKWyWX8fdG/j6UDYPbFog==",
+      "optional": true,
+      "requires": {
+        "@emnapi/wasi-threads": "1.0.1",
+        "tslib": "^2.4.0"
+      }
+    },
+    "@emnapi/runtime": {
+      "version": "1.3.1",
+      "resolved": "https://registry.npmjs.org/@emnapi/runtime/-/runtime-1.3.1.tgz",
+      "integrity": "sha512-kEBmG8KyqtxJZv+ygbEim+KCGtIq1fC22Ms3S4ziXmYKm8uyoLX0MHONVKwp+9opg390VaKRNt4a7A9NwmpNhw==",
+      "optional": true,
+      "requires": {
+        "tslib": "^2.4.0"
+      }
+    },
+    "@emnapi/wasi-threads": {
+      "version": "1.0.1",
+      "resolved": "https://registry.npmjs.org/@emnapi/wasi-threads/-/wasi-threads-1.0.1.tgz",
+      "integrity": "sha512-iIBu7mwkq4UQGeMEM8bLwNK962nXdhodeScX4slfQnRhEMMzvYivHhutCIk8uojvmASXXPC2WNEjwxFWk72Oqw==",
+      "optional": true,
+      "requires": {
+        "tslib": "^2.4.0"
```

Comment:
:thinking: Do you know why these `@emnapi`, and `@napi-rs` and `@node-rs` dependencies below are added here?

They're not specified in `easyops-cn/docusaurus-search-local`'s `package.json`, and `npm explain` wasn't helpful. Some of them are not installed in `node_modules`, but others like `@node-rs/jieba` are. That package pulls in some binaries related to https://github.com/messense/jieba-rs, which is a word segmentation package for Chinese implemented in Rust.

Ah, these are part of the `yarn.lock` file, which seems to also be used when resolving dependencies. The Node ecosystem is such a security nightmare... :face_exhaling:

It sucks that we have to depend on these packages even though we likely won't ever use Chinese in our docs. In general, I distrust Chinese software, perhaps unreasonably so, especially if it comes in binary blobs like this. Should we consider using [cmfcmf/docusaurus-search-local](https://github.com/cmfcmf/docusaurus-search-local) instead, which the `easyops-cn` package is a fork of? The changes from the fork don't seem that relevant for our use case.

Reply:
The reasons I chose `easyops-cn/docusaurus-search-local` over `cmfcmf/docusaurus-search-local` were mostly cosmetic. I think it looks a lot nicer but I'm also happy to try out `cmfcmf/docusaurus-search-local` and maybe tweak the design a bit 🤔

Reply:
Yeah, the `easyops-cn` package does look nicer. If the difference with the `cmfcmf` one is substantial, then let's not spend time tweaking the design. In all likelihood there's no reason to worry about the `easyops-cn` package, so let's keep it. It's just that seeing these binary deps does tingle my spidey senses :sweat_smile:, even if they're not part of the final bundle...